sky/provision/gcp/instance.py

```diff
@@ -244,3 +248,19 @@ def cleanup_ports(
         firewall_rule_name = provider_config['firewall_rule']
         instance_utils.GCPComputeInstance.delete_firewall_rule(
             project_id, firewall_rule_name)
+
+
+def query_ports(
```

Comment:
We should add this to `sky/provision/gcp/__init__.py` (Same for AWS and Azure):
```
from sky.provision.gcp.instance import query_ports
```

Otherwise `--endpoints` on a GCP cluster fails:
```
(base) ➜  sky-experiments git:(hd/k8s-ports) ✗ sky status test --endpoints
D 11-18 08:44:50 skypilot_config.py:158] Using config path: /Users/romilb/.sky/config.yaml
D 11-18 08:44:50 skypilot_config.py:161] Config loaded:
D 11-18 08:44:50 skypilot_config.py:161] None
D 11-18 08:44:50 skypilot_config.py:173] Config syntax check passed.
Traceback (most recent call last):
  File "/Users/romilb/tools/anaconda3/bin/sky", line 8, in <module>
    sys.exit(cli())
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 315, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1168, in invoke
    return super().invoke(ctx)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 336, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1947, in status
    raise ValueError('Querying endpoints is not supported '
ValueError: Querying endpoints is not supported for GCP.
```